plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -668,4 +669,17 @@ public static String fileName(String path)
     {
         return path.substring(path.lastIndexOf('/') + 1);
     }
+
+    public static void verifyNotPartitionedByNestedField(Schema schema, PartitionSpec partitionSpec)
+    {
+        Map<Integer, Integer> indexParents = indexParents(schema.asStruct());
+        partitionSpec.fields().forEach(field -> verifyNotPartitionedByNestedField(indexParents, field));
+    }
+
+    public static void verifyNotPartitionedByNestedField(Map<Integer, Integer> indexParents, PartitionField field)
+    {
+        if (indexParents.containsKey(field.sourceId())) {
```

Comment:
I'd be rather inclined to duplicate this check within `io.trino.plugin.iceberg.IcebergMetadata#getWriteLayout` then to make this method `public`. 
The `indexParents` map parameter is rather abstract.